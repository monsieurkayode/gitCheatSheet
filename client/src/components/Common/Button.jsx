import styled from 'styled-components';

export default styled.input`
  text-align: center;
  padding: 10px 40px;
  margin-left: 42px;
  min-width: 200px;
  background: #668090;
  color: #FFF;
  border-radius: 50px;
  font-size: 13px;
  font-weight: 500;
  font-family: Prompt, sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  letter-spacing: 2px;
  
  &:hover {
    background: #546975;
  }
`;
