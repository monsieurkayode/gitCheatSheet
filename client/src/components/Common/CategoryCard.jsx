import React from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

const Container = styled.div`
  height: 64px;
  flex: 1 1 23%;
  min-width: 160px;
  background: #668090;
  border-radius: 4px;
  margin: 5px;
  padding: 0 16px;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  text-transform: uppercase;
  font-family: Prompt, sans-serif;
  font-size: 13px;
  cursor: pointer;
  user-select: none;
  color: #fff;

  p {
    width: 100%;
  }

  .fiber {
    vertical-align: sub;
    font-size: 14px;
    color: #93d82a;
  }

  &:hover {
    background: #546975;
  }

  @media only screen and (max-width: 424px) {
    flex: 1 1 98%;
  }
`;

const CategoryCard = ({ category, history }) => (
  <Container
    onClick={() => history.push(`/cheats/${category.replace(/[ ]/g, '-')}`)}
    key={category}
    className="category"
  >
    <p>
      {category}&nbsp;&nbsp;
      <i className="material-icons fiber">fiber_manual_record</i>
    </p>
  </Container>
);

CategoryCard.propTypes = {
  category: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default withRouter(CategoryCard);
