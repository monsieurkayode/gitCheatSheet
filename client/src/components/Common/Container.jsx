import styled from 'styled-components';

export default styled.main`
  max-width: 740px;
  margin-top: 5%;
  margin-bottom: 60px;
  padding: 0 1%;
  flex: 1;
  width: 100%;
  z-index: 100;

  @media only screen and (max-width: 424px) {
    align-self: normal;
    margin-top: 10%;
    padding: 0 2%;
  }
`;
