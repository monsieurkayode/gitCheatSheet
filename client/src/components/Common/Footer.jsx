import React from 'react';
import styled from 'styled-components';

const Container = styled.footer`
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 740px;
  margin: 0 auto;
  padding: 0 8px;
  border-top: 1px solid rgba(0,0,0,0.1);
  padding: 16px;
  font-size: 14px;
  font-family: Prompt, sans-serif;
  width: 100%;
  user-select: none;

  p {
    span {
      background: #6756b3;
      color: #fff;
      padding: 4px 10px;
      border-radius: 15px;
      font-size: 12px;
    }
  }
`;

const Footer = () => (
  <Container>
    <p>&copy;&nbsp;2018</p>
    <p>Inspired By&nbsp;<span>devhints.io</span></p>
  </Container>
);

export default Footer;
