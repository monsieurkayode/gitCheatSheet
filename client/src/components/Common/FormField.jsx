import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';

const FieldWrapper = styled.div`
  .icon {
    font-size: 32px;
    -webkit-text-fill-color: #15234d;
    order: -1;
  }

  input[type="text"],
  input[type="password"] {
    width: 60%;
    margin-left: 10px;
    line-height: 32px;
    border: unset;
    border-bottom: 2px solid #668090;
    background: #f1f3f5;
    font-size: 14px;
    font-family: Roboto, sans-serif;

    &:focus {
      border-bottom: 2px solid #6756b3;
    }

    &.invalid {
      border-bottom: 2px solid crimson;
    }

    &.valid {
      border-bottom: 2px solid teal;
    }

    @media only screen and (max-width: 480px) {
      width: 80%;
    }
  }

  input:focus + .icon {
    -webkit-text-fill-color: #6756b3;
  }

  .error {
    width: 60%;
    font-size: 12px;
    color: crimson;
    font-family: Roboto, sans-serif;
    margin-top: 6px;
    text-align: right;
    transform: translateX(22px);

    @media only screen and (max-width: 480px) {
      width: 80%;
    }
  }
`;

const FormField = ({
  placeholder,
  icon,
  name,
  value,
  type,
  error
}) => (
  <FieldWrapper className="form-field">
    <input
      name={name}
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={() => {}}
      className={error.length > 0 ? 'invalid' : ''}
    />
    <i className="material-icons icon">{icon}</i>
    <div className="error">&nbsp;{error}</div>
  </FieldWrapper>
);

FormField.defaultProps = {
  value: '',
  type: 'text',
  error: ''
};

FormField.propTypes = {
  placeholder: string.isRequired,
  icon: string.isRequired,
  name: string.isRequired,
  value: string,
  type: string,
  error: string
};

export default FormField;
