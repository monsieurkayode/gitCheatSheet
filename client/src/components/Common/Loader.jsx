import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PreloaderIcon from 'react-preloader-icon';
import Oval from 'react-preloader-icon/loaders/Oval';

const LoaderWrapper = styled.div`
  &.loader {
    flex: ${({ loading }) => (loading ? 1 : 0)}
  }
`;
/**
 *
 * @function Loader
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const Loader = ({
  size,
  strokeColor,
  inline,
  loading
}) => (
  <LoaderWrapper
    style={{
      visibility: loading ? 'visible' : 'hidden'
    }}
    loading={loading}
    className={`${inline ? 'btn-loader' : 'loader'}`}
  >
    <PreloaderIcon
      loader={Oval}
      size={size}
      strokeWidth={12}
      strokeColor={strokeColor}
      duration={1200}
    />
  </LoaderWrapper>
);

Loader.defaultProps = {
  inline: false
};

Loader.propTypes = {
  size: PropTypes.number.isRequired,
  strokeColor: PropTypes.string.isRequired,
  inline: PropTypes.bool,
  loading: PropTypes.bool.isRequired
};

export default Loader;
