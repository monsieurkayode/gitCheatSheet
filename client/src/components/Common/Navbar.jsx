import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ReactTooltip from 'react-tooltip';
import { Link } from 'react-router-dom';
import Avatar from 'react-avatar';
import { toast } from 'react-toastify';
import { logoutUser } from '../../actions/userActions';
import logo from '../../assets/images/giticonw.png';

const Nav = styled.nav`
  height: 64px;
  line-height: 64px;
  text-align: center;
  position: relative;
  align-self: normal;

  @keyframes drop {
    from {
      margin-top: 10px;
      opacity: 0
    }
    to {
      margin-top: 30px;
      opacity: 0.9;
    }
  }

  .tooltip {
    line-height: initial;
    text-transform: uppercase;
    font-size: 10px;
    font-family: 'Josefin Sans', sans-serif;
    background: #668090;
    margin-top: 30px;
    box-shadow: 4px 3px 5px rgba(0,0,0,0.3);
    animation: drop .3s;
    visibility: unset;
    display: none;

    &.place-bottom::after {
      border-bottom-color: #668090;
    }

    &.place-left::after {
      border-left-color: #668090;
    }

    @media only screen and (max-width: 920px) {
      display: none;
    }
  }

  .show {
    visibility: unset;
    display: block;
  }

  .container {
    display: flex;
    align-items: center;
    max-width: 1232px;
    margin: 0 auto;
    padding: 0 1%;

    @media only screen and (min-width: 920px) and (max-width: 1260px) {
      padding: 0 3%;
    }

    .left {
      line-height: 0;

      img {
        width: 45px;
        min-width: 40px;
        min-height: 45px;
        background: #6756b3;
        border-radius: 50%;
        border: 2px solid #f1f3f5;
        padding: 10px;
        cursor: pointer;
      }
    }

    .brand {
      flex: 1 1 auto;
      cursor: pointer;
      
      > a {
        text-transform: uppercase;
        font-family:  'Josefin Sans', sans-serif;
        font-size: 13px;
        font-weight: 700;
        text-decoration: none;
        color: #15234d;

        &:hover {
          color: #745fb5;
        }
      }

    }

    .actions {
      font-family:  'Josefin Sans', sans-serif;
      min-width: 60px;

      ul {
        line-height: 0;
        list-style: none;
        padding: 0;
        margin: 0;

        li {
          display: inline-block;
          cursor: pointer;

          > i {
            font-size: 20px;
            -webkit-text-fill-color:  #15234d;
          }
        }

        > :last-child {
          padding-left: 20px;
        }
      }
    }
  }
`;

const AvatarDiv = styled.div`
  line-height: 0;
  display: ${props => (props.authenticated ? '' : 'none')}
`;

const NavMenu = styled.ul`
  display: ${props => (props.authenticated ? 'none' : '')}
`;

const NavBar = ({ user, authenticated, logout }) => (
  <Nav className="top-nav">
    <div className="container">
      <div className="left">
        <Link to="/" onClick={() => toast.dismiss()}>
          <img src={logo} alt="Logo" />
        </Link>
      </div>
      <div className="brand"><Link to="/">Gitcheatsheet</Link></div>
      <div className="actions">
        <AvatarDiv authenticated={authenticated} data-tip data-for="logout">
          <Avatar
            onClick={() => {
              ReactTooltip.hide();
              logout();
            }}
            style={{ cursor: 'pointer' }}
            round
            size={30}
            name={user.username}
            color="#668090"
          />
        </AvatarDiv>
        <NavMenu authenticated={authenticated}>
          <Link
            to="/register"
            onClick={() => {
              toast.dismiss();
              ReactTooltip.hide();
            }}
          >
            <li>
              <i data-tip data-for="register" className="material-icons">
                person_add
              </i>
            </li>
          </Link>
          <Link
            to="/login"
            onClick={() => {
              toast.dismiss();
              ReactTooltip.hide();
            }}
          >
            <li>
              <i data-tip data-for="login" className="material-icons">
                account_circle
              </i>
            </li>
          </Link>
        </NavMenu>
      </div>
    </div>
    <ReactTooltip className="tooltip" id="register" effect="solid">
      <span>Register</span>
    </ReactTooltip>
    <ReactTooltip className="tooltip" id="login" effect="solid">
      <span>Login</span>
    </ReactTooltip>
    <ReactTooltip className="tooltip" id="logout" effect="solid">
      <span>Logout</span>
    </ReactTooltip>
  </Nav>
);

NavBar.defaultProps = {
  user: {}
};

NavBar.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string,
    username: PropTypes.string
  }),
  logout: PropTypes.func.isRequired
};


const mapStateToProps = ({ userDetails }) => ({
  authenticated: Object.keys(userDetails.user).length === 2,
  user: userDetails.user
});

export default connect(mapStateToProps, { logout: logoutUser })(NavBar);
