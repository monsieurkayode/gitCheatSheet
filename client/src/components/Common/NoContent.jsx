import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  text-align: center;
  max-width: 1232px;
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  flex: 1;
  width: 100%;
  transform: translateY(-64px);

  i {
    color: #4d75a1;
    font-size: 50px;
  }
  
  .no-content-info {
    color: #668090;
    font-family: Roboto, sans-serif;
    font-weight: 400;
  }
`;

const NoContent = ({ content }) => (
  <Container className="no-content">
    <i className="material-icons">info</i>
    <h3 className="no-content-info">
      {content}
    </h3>
  </Container>
);

NoContent.propTypes = {
  content: PropTypes.string.isRequired
};

export default NoContent;
