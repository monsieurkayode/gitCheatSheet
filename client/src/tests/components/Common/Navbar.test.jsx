import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { NavBar, mapStateToProps } from '../../../components/Common/Navbar';

const setup = () => {
  const props = {
    authenticated: false,
    logout: jest.fn()
  };

  const wrapper = mount(
    <BrowserRouter>
      <NavBar {...props} />
    </BrowserRouter>
  );

  return { wrapper, props };
};

describe('NavBar Component', () => {
  const { wrapper, props } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call logout when the avatar is clicked', () => {
    const Avatar = wrapper.find('Avatar');
    Avatar.simulate('click');
    expect(props.logout).toHaveBeenCalled();
  });

  it('should trigger the click event for navigation links', () => {
    wrapper.find('Link').forEach(el => el.simulate('click'));
  });

  describe('mapStateToProps', () => {
    it('should return an object', () => {
      expect(mapStateToProps({ userDetails: { user: {} } }))
        .toBeInstanceOf(Object);
    });
  });
});
