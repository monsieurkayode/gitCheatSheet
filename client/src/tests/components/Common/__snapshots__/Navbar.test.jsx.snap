// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NavBar
      authenticated={false}
      logout={[MockFunction]}
      user={Object {}}
    >
      <styled.nav
        className="top-nav"
      >
        <StyledComponent
          className="top-nav"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "gPjMoj",
                "rules": Array [
                  "
  height: 64px;
  line-height: 64px;
  text-align: center;
  position: relative;
  align-self: normal;
  z-index: 10;

  @keyframes drop {
    from {
      margin-top: 10px;
      opacity: 0
    }
    to {
      margin-top: 30px;
      opacity: 0.9;
    }
  }

  .tooltip {
    line-height: initial;
    text-transform: uppercase;
    font-size: 10px;
    font-family: 'Josefin Sans', sans-serif;
    background: #668090;
    margin-top: 30px;
    box-shadow: 4px 3px 5px rgba(0,0,0,0.3);
    animation: drop .3s;
    visibility: unset;
    display: none;

    &.place-bottom::after {
      border-bottom-color: #668090;
    }

    &.place-left::after {
      border-left-color: #668090;
    }

    @media only screen and (max-width: 920px) {
      display: none;
    }
  }

  .show {
    visibility: unset;
    display: block;
  }

  .container {
    display: flex;
    align-items: center;
    max-width: 1232px;
    margin: 0 auto;
    padding: 0 1%;

    @media only screen and (min-width: 920px) and (max-width: 1260px) {
      padding: 0 3%;
    }

    .left {
      line-height: 0;

      img {
        width: 45px;
        min-width: 40px;
        min-height: 45px;
        background: #6756b3;
        border-radius: 50%;
        border: 2px solid #f1f3f5;
        padding: 10px;
        cursor: pointer;
      }
    }

    .brand {
      flex: 1 1 auto;
      cursor: pointer;
      
      > a {
        text-transform: uppercase;
        font-family:  'Josefin Sans', sans-serif;
        font-size: 13px;
        font-weight: 700;
        text-decoration: none;
        color: #15234d;

        &:hover {
          color: #745fb5;
        }
      }

    }

    .actions {
      font-family:  'Josefin Sans', sans-serif;
      min-width: 60px;

      ul {
        line-height: 0;
        list-style: none;
        padding: 0;
        margin: 0;

        li {
          display: inline-block;
          cursor: pointer;

          > i {
            font-size: 20px;
            -webkit-text-fill-color:  #15234d;
          }
        }

        > :last-child {
          padding-left: 20px;
        }
      }
    }
  }
",
                ],
              },
              "displayName": "styled.nav",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "nav",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <nav
            className="top-nav sc-bdVaJa gPjMoj"
          >
            <div
              className="container"
            >
              <div
                className="left"
              >
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <img
                      alt="Logo"
                      src="test-file-mock"
                    />
                  </a>
                </Link>
              </div>
              <div
                className="brand"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    Gitcheatsheet
                  </a>
                </Link>
              </div>
              <div
                className="actions"
              >
                <styled.div
                  authenticated={false}
                  data-for="logout"
                  data-tip={true}
                >
                  <StyledComponent
                    authenticated={false}
                    data-for="logout"
                    data-tip={true}
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": false,
                          "lastClassName": "iGxHnd",
                          "rules": Array [
                            "
  line-height: 0;
  display: ",
                            [Function],
                            "
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bwzfXH iGxHnd"
                      data-for="logout"
                      data-tip={true}
                    >
                      <withAvatarConfig
                        color="#668090"
                        onClick={[Function]}
                        round={true}
                        size={30}
                        style={
                          Object {
                            "cursor": "pointer",
                          }
                        }
                      >
                        <Avatar
                          avatarRedirectUrl={null}
                          cache={
                            Object {
                              "get": [Function],
                              "hasSourceFailedBefore": [Function],
                              "set": [Function],
                              "sourceFailed": [Function],
                            }
                          }
                          className=""
                          color="#668090"
                          colors={
                            Array [
                              "#d73d32",
                              "#7e3794",
                              "#4285f4",
                              "#67ae3f",
                              "#d61a7f",
                              "#ff4080",
                            ]
                          }
                          fgColor="#FFF"
                          initials={[Function]}
                          onClick={[Function]}
                          round={true}
                          size={30}
                          style={
                            Object {
                              "cursor": "pointer",
                            }
                          }
                          textSizeRatio={3}
                          unstyled={false}
                        >
                          <div
                            className=" sb-avatar"
                            onClick={[Function]}
                            style={
                              Object {
                                "borderRadius": "100%",
                                "cursor": "pointer",
                                "display": "inline-block",
                                "fontFamily": "Helvetica, Arial, sans-serif",
                                "height": "30px",
                                "verticalAlign": "middle",
                                "width": "30px",
                              }
                            }
                          >
                            <div
                              className=" sb-avatar__text"
                              style={
                                Object {
                                  "background": "#668090",
                                  "borderRadius": "100%",
                                  "color": "#FFF",
                                  "height": "30px",
                                  "lineHeight": "initial",
                                  "textAlign": "center",
                                  "textTransform": "uppercase",
                                  "width": "30px",
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "display": "table",
                                    "height": "100%",
                                    "width": "100%",
                                  }
                                }
                              >
                                <span
                                  style={
                                    Object {
                                      "display": "table-cell",
                                      "verticalAlign": "middle",
                                    }
                                  }
                                >
                                  <span
                                    key="null"
                                  />
                                </span>
                              </div>
                            </div>
                          </div>
                        </Avatar>
                      </withAvatarConfig>
                    </div>
                  </StyledComponent>
                </styled.div>
                <styled.ul
                  authenticated={false}
                >
                  <StyledComponent
                    authenticated={false}
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "lastClassName": "dyPnRr",
                          "rules": Array [
                            "
  display: ",
                            [Function],
                            "
",
                          ],
                        },
                        "displayName": "styled.ul",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "ul",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <ul
                      className="sc-htpNat dyPnRr"
                    >
                      <Link
                        onClick={[Function]}
                        replace={false}
                        to="/register"
                      >
                        <a
                          href="/register"
                          onClick={[Function]}
                        >
                          <li>
                            <i
                              className="material-icons"
                              data-for="register"
                              data-tip={true}
                            >
                              person_add
                            </i>
                          </li>
                        </a>
                      </Link>
                      <Link
                        onClick={[Function]}
                        replace={false}
                        to="/login"
                      >
                        <a
                          href="/login"
                          onClick={[Function]}
                        >
                          <li>
                            <i
                              className="material-icons"
                              data-for="login"
                              data-tip={true}
                            >
                              account_circle
                            </i>
                          </li>
                        </a>
                      </Link>
                    </ul>
                  </StyledComponent>
                </styled.ul>
              </div>
            </div>
            <ReactTooltip
              className="tooltip"
              effect="solid"
              id="register"
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
                id="register"
              >
                <span>
                  Register
                </span>
              </div>
            </ReactTooltip>
            <ReactTooltip
              className="tooltip"
              effect="solid"
              id="login"
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
                id="login"
              >
                <span>
                  Login
                </span>
              </div>
            </ReactTooltip>
            <ReactTooltip
              className="tooltip"
              effect="solid"
              id="logout"
              insecure={true}
              resizeHide={true}
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
                id="logout"
              >
                <span>
                  Logout
                </span>
              </div>
            </ReactTooltip>
          </nav>
        </StyledComponent>
      </styled.nav>
    </NavBar>
  </Router>
</BrowserRouter>
`;
